modules/ServerAPI/src/client/Protocol.coffee

```diff
@@ -499,3 +530,560 @@ class ProtocolBaseController extends BaseEntityController
+		#hide previously shown warnings/success text
+		$(".bv_downloadSuccess").hide()
+		$(".bv_downloadWarning").hide()
+
+		protocolCode = @model.escape('codeName')		
+		$.ajax
+			type: 'GET'
+			#there are two similar routes in ExperimentBrowserRoutes.coffee and ExperimentServiceRoutes.coffee
+			#url: "/api/experimentsForProtocol/#{protocolCode}" #ExperimentBrowserRoutes.coffee route
+			url: "/api/experiments/protocolCodename/#{protocolCode}" #ExperimentServiceRoutes.coffee route
+			success: (experiments) =>
+				@setupExperimentSummaryTable experiments
+				$(".bv_experimentTableControllerTitle").html "Experiments using " + protocolCode + ":"
+
+
+	handleEndpointRowPressed: =>
+		if @options.readOnly == false #disable logic if endpoint list controller is read only (since there won't be an experiment controller)
+			#When an endpoint row is pressed, we want to filter the experiment summary table so it only displays the experiments...
+			#...with protocols that contain the endpoint, so we'll need to find the row's endpoint name and regenerate the table
+
+			#since any element within the row could be clicked on, we need to first find the parent row div and move down from there
+			element = $(event.target)
+			for x in [0..20] #allow it to search up to 20 parent elements up for the table row (usually takes 10 or less)
+				#if the element is a table row, exit the loop
+				if element[0].nodeName == "TR"
+					tr = element
+					break
+				#if the element is not a table row, move up to the next parent element
+				else
+					element = element.parent()
+			
+			#we need to detect if the element we have clicked on is the previously selected element
+			if "selectedEndpointRow" in tr[0].classList
+				previouslySelectedRow = true
+			else
+				previouslySelectedRow = false
+
+			#first reset the background color of all the rows
+			for elm in tr.parent()[0].childNodes
+				for subelm in elm.childNodes
+					try
+						subelm.style.background = "#F9F9F9"
+					catch
+						#not all elements can be styled, so do nothing
+
+			#remove class marking whether a row was previously selected 
+			$(".selectedEndpointRow").removeClass "selectedEndpointRow"
+
+
+			#if the row was previously selected, load table for just experiments associated with protocol, no filtering by endpoint
+			if previouslySelectedRow == true
+				@getExperimentSummaryTable()
+		
+			#if the row was not previously selected, load table with associated experiments filtered by endpoint and highlight endpoint row
+			else
+				#mark the selected row
+				tr.addClass('selectedEndpointRow')
+
+				#apply highlighting to the selected rows
+				for trElements in tr
+					for td in trElements.childNodes
+						try
+							td.style.background = "#D9EDF7"
+						catch
+							#not all elements can be styled, so do nothing
+
+				#once the row is found, extract the endpoint values from it from it
+				endpointRowValues = tr[0].querySelectorAll("span.select2-selection__rendered")
```

Comment:
nit: I can see how this works to get the text from the row, but this part feels a little indirect. Is there a way we could look up the LsState that corresponds to the clicked row, and more directly get the LsValues within?

This would be a larger change, but if you see how `EndpointController.removeRow` works, it uses the context of "which endpoint row was clicked" to know the rowNumber, then passes that up to the EndpointListController.
I could imagine you doing the same thing here, which would give you direct access to the LsState in question, and avoid having to grab the text and then translate it back to the "code"

Reply:
@bffrost That makes a lot of sense! I'm not sure how to pass that up to the EndpointListController though?

Reply:
@hhan-schrodinger to pass it up, you'd do something like we do at the end of `EndpointController.removeRow`:
```
		# Alert the parent controller to destroy this controller
		@trigger 'rowRemoved', @rowNumber
```

That will fire an event, and then we set up the EndpointListController to listen for it:
```
class EndpointListController extends AbstractFormController
	template: _.template($("#EndpointListView").html())

	events:
		"click .bv_addEndpoint": "handleAddEndpointPressed"
		"rowRemoved": "handleRowRemoved" <--- here
		"click .bv_endpointRow": "handleEndpointRowPressed"
		"click .bv_downloadFiles": "downloadFiles"
```

And then the `handleRowRemoved` function takes an input which is the rowNumber:
```
	handleRowRemoved: (rowNumber) =>
		if @endpointControllers[rowNumber]?
			@endpointControllers[rowNumber].remove()
			@endpointControllers[rowNumber].unbind()
			@endpointControllers[rowNumber].el.remove()
```